handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -775,9 +784,29 @@ public final SSLEngineResult unwrap(
                 }
             }
 
-            // Write encrypted data to network BIO
+            if (len < SslUtils.SSL_RECORD_HEADER_LENGTH) {
+                return new SSLEngineResult(BUFFER_UNDERFLOW, getHandshakeStatus(), 0, 0);
+            }
+
+            int packetLength = SslUtils.getEncryptedPacketLength(srcs, srcsOffset);
```

Comment:
@normanmaurer Have you verified that doing this each time will not impact performance? SslHandler would cache the packetLength so that it wouldn't recalculate it each time. I suspect it's not an issue.